test/fixtures/cookbooks/install_varnish/templates/default/lib_default_custom.vcl.erb

```diff
@@ -0,0 +1,14 @@
+<%- if @varnish_version == '4' %>
+vcl 4.0;
+<%- end %>
+backend default {
+        .host = "<%= @config.backend_host %>";
+        .port = "<%= @config.backend_port %>";
+        .probe = {
+          .url = "<%= @parameters[:url] %>";
```

Comment:
It would yes as `@parameters[:url]` would be `nil`, but since this would be a custom template in a wrapper cookbook and any parameters set would be set (or not set) by the author/users of that wrapper cookbook, I'd argue that it would be their concern and out of scope for the varnish cookbook. It's merely an example to exercise all 3 of the new resource attributes for ChefSpec, however clearly if the author of a custom template was concerned about other users of that template not passing in all the hash keys used by it they could `<% @parameters[:url] ||= '/' %>` at the top of the template or similar.